src/client/store/ui/fileUpload/actions/uploadFiles.ts

```diff
@@ -0,0 +1,28 @@
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import axios from 'axios'
+
+import { ApiEndPoint } from 'meta/api/endpoint'
+import { CycleParams } from 'meta/api/request'
+import { File as FileType } from 'meta/file'
+
+import { ThunkApiConfig } from 'client/store/types'
+
+type Props = CycleParams & {
+  files: Array<File>
+}
+
+export const uploadFiles = createAsyncThunk<Array<FileType>, Props, ThunkApiConfig>(
+  'fileUpload/uploadFiles',
+  async (props) => {
+    const { assessmentName, cycleName, countryIso, files } = props
+    const formData = new FormData()
+    files.forEach((file) => formData.append('file', file))
```

Comment:
note:
add many files under same key 'file' in formData which is handled by multer in backend to create an array of files